20241014-wordcloud/src/routes/+page.svelte

```diff
@@ -0,0 +1,170 @@
+<script lang="ts">
+    import { onMount } from "svelte";
+    import WordCloud from "./WordCloud.svelte";
+
+    let words = [
+        {text: "placeholder", count: 1},
+    ];
+    let submission: string = "";
+    let submissionInput: HTMLInputElement;
+    const maxTimer = 5;
+
+    function invalidate(placeholder: string) {
+        submissionInput.placeholder = placeholder;
+        submissionInput.style.transition = "";
+        submissionInput.value = "";
+        submissionInput.style.backgroundColor = "#ffccdb";
+        setTimeout(() => {
+            submissionInput.style.transition = "background-color 0.5s";
+            submissionInput.style.backgroundColor = "white";
+        }, 100);
+        setTimeout(() => {
+            submissionInput.placeholder = "Enter a word...";
+        }, 600);
+    }
+
+    async function submitWord() {
+        const rgx = /drop\s+table/i;
+        if (rgx.test(submission)) {
+            alert("Nice try, but my SQL inputs are thoroughly sanitised.");
+        }
+
+        if (submission === "") {
+            return;
+        }
+
+        // Sanitise input
+        submission = submission
+            .toLowerCase()
+            .trim()
+            .split(/\s+/).join("-")
+            .replace(/[^a-z0-9-_.]/g, "");
+
+        if (submission === "") {
+            invalidate("invalid character in input");
+            return;
+        }
+
+        if (localStorage.getItem(`wordcloud___${submission}`) === "true") {
+            invalidate("already submitted");
+            return;
+        } else {
+            localStorage.setItem(`wordcloud___${submission}`, "true");
+        }
+
+        const response = await fetch("/answer", {
+            method: "POST",
+            headers: {
+                "Content-Type": "application/json",
+            },
+            body: JSON.stringify({text: submission}),
+        });
+        words = await response.json();
+        timer = maxTimer;
+        submission = "";
+    }
+
+    async function getWords() {
+        const response = await fetch("/ask");
+        words = await response.json();
+        // Print to console so that it's easy to get hold of the data if we need to replot
+        let s = [];
+        for (const {text, count} of words) {
+            for (let i = 0; i < count; i++) {
+                s.push(text);
+            }
+        }
+        console.log(s.join(" "));
+    }
+
+    let timer = maxTimer;
+
+    onMount(async () => {
+        await getWords();
+        setInterval(async () => {
+            timer -= 1;
+            if (timer <= 0) {
+                await getWords();
+                timer = maxTimer;
+            }
+        }, 1000);
```

Comment:
This `1000` feels a bit arbitrary. Maybe store it in a variable, with a name that describes its purpose?